packages/permission-common/src/types/permission.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * The basic operation being performed in relation to this permission, expressed as a CRUD action.
+ * @public
+ */
+export enum PermissionAction {
```

Comment:
Tbh you may want to avoid enums, they tend to just make things complicated and verbose without helping too much 😅

One enum-less option would be a set of const `PERMISSION_ACTION_CREATE` etc, along with `type PermissionAction = 'create' | 'read' ...`. Constants could even be skipped too given that TS works pretty well with just string constants.

Very much just a somewhat weakly held opinion though, so do feel free to disregard or convince me otherwise :grin:

Reply:
Union of string constants seems fine here, I agree. I ditched this one but left `AuthorizationResult` as an enum, this will be referenced quite a bit in other code and `AuthorizationResult.ALLOW` somehow feels nicer when it's used often than just a string floating around everywhere 🤔